src/lib.rs

```diff
@@ -527,13 +538,52 @@ impl SimulationSpace {
                         Attribute::POSITION.name()
                     ))
                 }
+                SimulationSpace::GlobalWithOffset => {
```

Comment:
The init is identical to Global at this point since adding the property value and subtracting the particle's offset value would cancel each other out in the initial spawn/clone phase when these values are identical.
However I have kept a separate section here to add the verification of attribute & property layouts.